x/oracle/types/msgs.go

```diff
@@ -145,8 +147,15 @@ func (msg MsgAggregateExchangeRateVote) ValidateBasic() error {
 		}
 	}
 
-	if len(msg.Salt) > 4 || len(msg.Salt) < 1 {
-		return sdkerrors.Wrap(ErrInvalidSaltLength, "salt length must be [1, 4]")
+	if len(msg.Salt) > 80 || len(msg.Salt) < 60 {
+		return ErrInvalidSaltLength
+	}
+	_, err = AggregateVoteHashFromHexString(msg.Salt)
+	if err != nil {
+		return sdkerrors.Wrap(ErrInvalidSaltFormat, "salt must be a valid hex string")
+	}
+	if strings.Contains(msg.Salt, ":") {
```

Comment:
The `AggregateVoteHashFromHexString` check makes this redundant - encoding/hex/hex.go performs this check : 
```go
// fromHexChar converts a hex character into its value and a success flag.
func fromHexChar(c byte) (byte, bool) {
	switch {
	case '0' <= c && c <= '9':
		return c - '0', true
	case 'a' <= c && c <= 'f':
		return c - 'a' + 10, true
	case 'A' <= c && c <= 'F':
		return c - 'A' + 10, true
	}

	return 0, false
}

I'll take this out and just keep the colon char test in
```